_posts/en/newsletters/2023-02-08-newsletter.md

```diff
@@ -0,0 +1,199 @@
+It does this by enhancing `AddrMan`
+(the peer address manager) to keep track of the number of address entries
+separately by network and "tried" versus "new" type, which in turn allows
+better use of fixed seeds. This is the first step in a larger effort to
+improve outbound peer selection.
+
+{% include functions/details-list.md
+  q0="When is a network considered reachable?"
+  a0="A network is assumed to be reachable unless we're sure we
+      can't access it, or our configuration has specified one or
+      more _other_ networks using the `-onlynet=` configuration option
+      (only those are considered reachable, even if other network
+      types are actually available).
+  a0link="https://bitcoincore.reviews/26847#l-22"
+
+  q1="How is an address received over the p2p network treated depending
+      on whether the address's network is reachable vs. non-reachable --
+      do we store it (add it to `AddrMan`) and/or forward it to peers?"
+  a1="If its network is reachable, we relay the address to two
+      randomly-chosen peers, else we relay it to 1 or 2 peers (whether
+      1 or 2 is randomly-chosen).
+      We only store the address if its network is reachable."
+  a1link="https://bitcoincore.reviews/26847#l-51"
+
+  q2="How can a node currently get stuck with only unreachable addresses
+      in `AddrMan`, finding no outbound peers? How does this PR fix it?"
+  a2="If the `-onlynet` configuration changes. For example, suppose the
+      node has always been run with `-onlynet=ipv4` so its AddMan has
+      no ipv6 addresses. Then the node is restarted with `-onlynet=onion`.
+      The fixed seeds have some onion addresses, but without the PR, the node
+      won't use them since the `AddrMan` isn't _completely_ empty (it has some
+      ipv4 addresses from before). With the PR, the startup code will add
+      some onion fixed seeds, since `AddrMan` doesn't contain any addresses
+      of _that_ (now reachable) network type.
+  a2link="https://bitcoincore.reviews/26847#l-98"
+
+  q3="When an address we'd like to add to `AddrMan` collides with an existing
+      address, what happens? Is the existing address always dropped in favor
+      of the new address?"
+  a3="No, generally the existing address is retained (not the new one),
+      unless the existing address is deemed "terrible"
+      (see `AddrInfo::IsTerrible()`).
+  a3link="https://bitcoincore.reviews/26847#l-100"
+
+  q4="Why would it be beneficial to have an outbound connection to each
+      reachable network at all times?"
+  a4="A selfish reason is that it's harder to eclipse-attack the node,
+      since the attacker would need to run nodes on multiple networks.
+      A non-selfish reason is that it helps keep the overall network
+      together, avoiding chain splits caused by network partitions.
+      If half the nodes, including miners, ran with `-onlynet=x` and
+      the other half, including miners, ran `-onlynet=y`, then two
+      chains could emerge."
```

Comment:
Would note that these positions are gradually changing.  There are currently around 14k Tor peers and 1k I2P peers.  Also, from discussions with node operators and as recommended by Bitcoin Core documentation, those who are using onlynet are increasingly doing so with multiple networks.

Reply:
That's interesting, but I think a little beyond the point I was trying to make here (about possible chain splits). Also, I don't want to stray too far from what was discussed during the review club meeting (this is supposed to be just a summary of the meeting).

Reply:
It should probably also be noted that it's possible to have an outbound connection to each reachable network at all times simply by adding manual connections to your conf file.

Reply:
That's really good to know, and I hadn't thought of that myself, but again I'm not sure it's appropriate to add things here that weren't covered in the review club meeting at all. If it _is_ okay to do that, let me know, and I'd be glad to add it!

Reply:
It's ok to add things that weren't covered during the meeting if they make the material more useful.